lib/src/se/llbit/chunky/ui/SizeInput.java

```diff
@@ -0,0 +1,300 @@
+      listenerLockedByUpdate.set(true);
+      queuedHeightValueProperty().set(getHeightValue());
+      listenerLockedByUpdate.set(false);
+    });
+
+    widthInput.setOnAction(event -> submitChanges());
+    heightInput.setOnAction(event -> submitChanges());
+
+    showAspectRatioDetailsProperty().addListener((observable, oldValue, newValue) -> {
+      if(newValue != oldValue) {
+        if(newValue) {
+          getChildren().add(aspectRatioDetailsNode);
+        } else {
+          getChildren().remove(aspectRatioDetailsNode);
+        }
+      }
+    });
+  }
+
+  private Node buildAspectRatioDetails() {
+    Label details = new Label();
+    details.setPadding(new Insets(1, 0, 2, 3));
+    details.setStyle("-fx-text-fill: #999; -fx-font-style: italic;");
+    details.visibleProperty().bind(showAspectRatioDetailsProperty);
+
+    IntegerProperty gdc = new SimpleIntegerProperty(QuickMath.gcd(getWidthValue(), getHeightValue()));
+
+    details.textProperty().bind(
+      Bindings.concat(
+        widthValueProperty(), unitStringProperty(),
+        " × ",
+        heightValueProperty(), unitStringProperty(),
+        " (",
+        Bindings.createIntegerBinding(
+          () -> (int) Math.ceil((double) getWidthValue() / gdc.get()),
+          gdc, widthValueProperty()
+        ),
+        " : ",
+        Bindings.createIntegerBinding(
+          () -> (int) Math.ceil((double) getHeightValue() / gdc.get()),
+          gdc, heightValueProperty()
+        ),
+        ")"
+      )
+    );
+    changeQueuedProperty().addListener((observable, oldValue, newValue) -> {
+      // size changes were committed
+      if(!newValue) {
+        gdc.set(QuickMath.gcd(getWidthValue(), getHeightValue()));
+      }
+    });
+    return details;
+  }
+
+  public IntegerProperty widthValueProperty() {
+    return currentWidthProperty;
+  }
+
+  public IntegerProperty heightValueProperty() {
+    return currentHeightProperty;
+  }
+
+  public IntegerProperty queuedWidthValueProperty() {
+    return widthInput.valueProperty();
+  }
+
+  public IntegerProperty queuedHeightValueProperty() {
+    return heightInput.valueProperty();
+  }
+
+  public BooleanProperty ratioLockedProperty() {
+    return ratioLockCheckBox.selectedProperty();
+  }
+
+  public StringProperty unitStringProperty() {
+    return unitLabel.textProperty();
+  }
+
+  public ReadOnlyBooleanProperty changeQueuedProperty() {
+    return changeQueuedProperty;
+  }
+
+  public BooleanProperty showAspectRatioDetailsProperty() {
+    return showAspectRatioDetailsProperty;
+  }
+
+  public boolean isRatioLocked() {
+    return ratioLockedProperty().get();
+  }
+
+  public int getWidthValue() {
+    return widthValueProperty().get();
+  }
+
+  public int getHeightValue() {
+    return heightValueProperty().get();
+  }
+
+  public int getQueuedWidthValue() {
+    return queuedWidthValueProperty().get();
+  }
+
+  public int getQueuedHeightValue() {
+    return queuedHeightValueProperty().get();
+  }
+
```

Comment:
The properties are already exposed, is there a need to expose additional getters?

```suggestion
```

Reply:
Other JavaFX nodes do expose them for convenience reasons - now these methods have JavaDocs to give them a reason to exist 😄